browser/ui/webui/ai_chat/ai_chat_untrusted_conversation_ui.cc

```diff
@@ -0,0 +1,211 @@
+
+    AIChatUI* ai_chat_ui_controller =
+        rfh->GetWebUI()->GetController()->GetAs<AIChatUI>();
+    // We should not be embedded on any non AIChatUI page
+    CHECK(ai_chat_ui_controller);
+
+    ai_chat_ui_controller->BindInterface(std::move(parent_ui_frame_receiver));
+  }
+
+ private:
+  void OpenURL(GURL url) {
+    if (!url.SchemeIs(url::kHttpsScheme)) {
+      return;
+    }
+
+#if !BUILDFLAG(IS_ANDROID)
+    Browser* browser = ai_chat::GetBrowserForWebContents(owner_web_contents_);
+    browser->OpenURL(
+        {url, content::Referrer(), WindowOpenDisposition::NEW_FOREGROUND_TAB,
+         ui::PAGE_TRANSITION_LINK, false},
+        /*navigation_handle_callback=*/{});
+#else
+    // We handle open link different on Android as we need to close the chat
+    // window because it's always full screen
+    ai_chat::OpenURL(url.spec());
+#endif
+  }
+
+  raw_ptr<content::WebContents> owner_web_contents_ = nullptr;
+  mojo::Receiver<ai_chat::mojom::UntrustedUIHandler> receiver_;
+};
+
+}  // namespace
+
+bool AIChatUntrustedConversationUIConfig::IsWebUIEnabled(
+    content::BrowserContext* browser_context) {
+  // Only enabled if we have a valid service
+  return (ai_chat::AIChatServiceFactory::GetForBrowserContext(
+              browser_context) != nullptr);
+}
+
+std::unique_ptr<content::WebUIController>
+AIChatUntrustedConversationUIConfig::CreateWebUIController(
+    content::WebUI* web_ui,
+    const GURL& url) {
+  return std::make_unique<AIChatUntrustedConversationUI>(web_ui);
+}
+
+AIChatUntrustedConversationUIConfig::AIChatUntrustedConversationUIConfig()
+    : WebUIConfig(content::kChromeUIUntrustedScheme,
+                  kAIChatUntrustedConversationUIHost) {}
+
+AIChatUntrustedConversationUIConfig::~AIChatUntrustedConversationUIConfig() =
+    default;
+
+AIChatUntrustedConversationUI::AIChatUntrustedConversationUI(
+    content::WebUI* web_ui)
+    : ui::MojoWebUIController(web_ui) {
+  // Create a URLDataSource and add resources.
+  content::WebUIDataSource* source = content::WebUIDataSource::CreateAndAdd(
+      web_ui->GetWebContents()->GetBrowserContext(),
+      kAIChatUntrustedConversationUIURL);
+  webui::SetupWebUIDataSource(source, kAiChatUiGenerated,
+                              IDR_AI_CHAT_UNTRUSTED_CONVERSATION_UI_HTML);
+
+  for (const auto& str : ai_chat::GetLocalizedStrings()) {
+    source->AddString(str.name,
+                      brave_l10n::GetLocalizedResourceUTF16String(str.id));
+  }
+
+#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS)
+  constexpr bool kIsMobile = true;
+#else
+  constexpr bool kIsMobile = false;
+#endif
+  source->AddBoolean("isMobile", kIsMobile);
+
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ScriptSrc,
+      "script-src 'self' chrome-untrusted://resources;");
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::StyleSrc,
+      "style-src 'self' 'unsafe-inline' chrome-untrusted://resources;");
```

Comment:
do we need 'unsafe-inline'? opens us up to style injection attacks

Reply:
style-src 'unsafe-inline' is used by all Chromium WebUI (at least it is by the Settings WebUI) for adding CSS dynamically to the page for the web components.